### PR TITLE
no injection mock.patch helper

### PR DESCRIPTION
**Pull request by jonapich**
Implement a decorator that prevents a mock from being injected into the test function.

I need suggestions for naming things.

### CODE REVIEW

File: coveo-testing/coveo_testing/noyield.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import functools
+from typing import Callable, Any
+from unittest.mock import _patch
+
+
+class no_yield:  # noqa: lowercased decorator
+    """
+    Discards the yield return value of a `mock.patch` fixture. Usage:
+
+        @void(mock.patch(...))
```
Comment (wtrep) [inline coveo-testing/coveo_testing/noyield.py:10]: void ?

File: coveo-testing/coveo_testing/noyield.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import functools
+from typing import Callable, Any
+from unittest.mock import _patch
+
+
+class no_yield:  # noqa: lowercased decorator
```
Comment (wtrep) [inline coveo-testing/coveo_testing/noyield.py:6]: I'm fine with `@no_yield`. `@without_yield` could also be an option

**Review by dotboris — APPROVED** [review]

**Comment by jonapich** [comment]
After more testing, I found out there were too many edge cases making the feature work only  in some specific situations. For instance, the behavior can be broken by:

- The order of decorators
- Decorators that forgot to `functools.wraps()`
- Test-patch vs fixture-patch
- yield-fixtures vs return-fixtures
